app/assets/javascripts/discourse/app/lib/utilities.js

```diff
@@ -799,3 +799,7 @@ export function isPrimaryTab() {
     }
   });
 }
+
+export function optionalRequire(path) {
+  return require.has(path) && require(path).default;
```

Comment:
What if someone wants to optionalImport a named export? I think we should probably drop the `.default` here, and then consumers can do `const MyComponent = optionalRequire(path)?.default`,

Reply:
I considered that, but then there were no use cases like that so decided to go with this version. I was thinking of this variant:

```js
export function optionalRequire(path, name = "default") {
  return require.has(path) && require(path)[name];
```

so the most common case can stay the same, but you can require a non-default import.

Reply:
Could we add that `name=` parameter here, so that the intention is clear? Otherwise people might start doing the `require.has` stuff themselves.